### ISSUE TITLE
Display mixed line endings explicitly in status bar as "LF (mixed)" or "CRLF (mixed)"

### ISSUE THREAD

**Issue by geekley** [microsoft/vscode#119025]
When opening a file with mixed line endings, vscode is still showing it as `LF` or `CRLF` in the status bar.
After saving, it fixes it, but I think it should show, for example, `LF (mixed)` (maybe with squiggle or :warning: icon) until the file is modified.
This is important so that you know the file still has mixed EOLs, and you might want to fix that.

To reproduce: `printf "lf\ncrlf\r\ncr\rnone" > mixed-eol.txt && code mixed-eol.txt`
If you open this file in Notepad++ and enable visible EOL, you can see it has mixed endings.

(Posted from https://github.com/microsoft/vscode/issues/12223#issuecomment-391511403 as separate issue.)

**Comment by geekley** [comment]
You might also want to display a confirmation dialog when saving, before auto-fixing the mixed line-endings, just to avoid unexpected/unwanted automatic behavior when user is just doing a simple edit operation (specially if they happen to open a binary file as text).

The confirmation dialog could have buttons `Use LF`, `Use CRLF`, `Leave mixed` and `Cancel`, so that the user makes an explicit choice in this case. The main/primary button may be the one used the most on the file. If the user chooses `Leave mixed`, the file is saved retaining mixed EOL (only new lines are added like shown in status bar), and that choice is remembered for that file until session ends.

**Comment by microestc** [comment]
Yes, I'm sure it's useful. a file contains a LF CRLF mix of cases, and I don't want to convert to a particular one in exceptional cases

**Comment by microestc** [comment]
When a file contains a mixture of lf and crlf, I don't want to change any lf or crlf changes when I save and format the code, but now the reality is that it changes automatically

**Comment by microestc** [comment]
When a file contains a mixture of lf and crlf, I don't want to change any lf or crlf changes when I save and format the code, but now the reality is that it changes automatically

**Comment by sujingang** [comment]
> 当文件包含lf和crlf的混合时，我不想在保存和格式化代码时更改任何lf或crlf更改，但现在的现实是它会自动更改

I also find it useful, especially when it comes to git code, always thinking that I have changed a lot of places.

**Comment by AriESQ** [comment]
Not having the ability to show newlines, CR/LF, EOF is bad for both beginners and experienced developers. I mention this because there is a lot of positioning of vScode as a solution for beginners.

**Comment by nic-r** [comment]
any plan on this feature

**Comment by mcrossley** [comment]
Still nothing?

**Comment by mcrossley** [comment]
I'd still really like this. It is a feature of Visual Studio I really miss in VSC
